components/script/dom/htmlcanvaselement.rs

```diff
@@ -60,6 +65,18 @@ impl HTMLCanvasElement {
         let element = HTMLCanvasElement::new_inherited(localName, prefix, document);
         Node::reflect_node(box element, document, HTMLCanvasElementBinding::Wrap)
     }
+
+    fn recreate_contexts(&self) {
+        let (w, h) = (self.width.get() as i32, self.height.get() as i32);
```

Comment:
I've seen this in several places. Why not factor out this into an `fn size()` that returns a `Size2D<i32>`?